cmake/fake_tribits.cmake

```diff
@@ -0,0 +1,344 @@
+      ""
+      "EXE;NAME"
+      ""
+      ${ARGN})
+    IF(TEST_EXE)
+      SET(EXE_ROOT ${TEST_EXE})
+    ELSE()
+      SET(EXE_ROOT ${TEST_NAME})
+    ENDIF()
+
+    TRIBITS_ADD_TEST(
+      ${EXE_ROOT}
+      NAME ${TEST_NAME}
+      ${ARGN} 
+      COMM serial mpi
+      NUM_MPI_PROCS 1
+      ${TEST_UNPARSED_ARGUMENTS}
+    )
+  else()
+    CMAKE_PARSE_ARGUMENTS(TEST 
+      "WILL_FAIL"
+      "FAIL_REGULAR_EXPRESSION;PASS_REGULAR_EXPRESSION;EXE;NAME"
+      "CATEGORIES"
+      ${ARGN})
+    IF(TEST_EXE)
+      SET(EXE ${TEST_EXE})
+    ELSE()
+      SET(EXE ${TEST_NAME})
+    ENDIF()
+    IF(WIN32)
+      ADD_TEST(NAME ${TEST_NAME} WORKING_DIRECTORY ${LIBRARY_OUTPUT_PATH} COMMAND ${EXE}${CMAKE_EXECUTABLE_SUFFIX})
+    ELSE()
+      ADD_TEST(NAME ${TEST_NAME} COMMAND ${EXE})
+    ENDIF()
+    IF(TEST_WILL_FAIL)
+      SET_TESTS_PROPERTIES(${TEST_NAME} PROPERTIES WILL_FAIL ${TEST_WILL_FAIL})
+    ENDIF()
+    IF(TEST_FAIL_REGULAR_EXPRESSION)
+      SET_TESTS_PROPERTIES(${TEST_NAME} PROPERTIES FAIL_REGULAR_EXPRESSION ${TEST_FAIL_REGULAR_EXPRESSION})
+    ENDIF()
+    IF(TEST_PASS_REGULAR_EXPRESSION)
+      SET_TESTS_PROPERTIES(${TEST_NAME} PROPERTIES PASS_REGULAR_EXPRESSION ${TEST_PASS_REGULAR_EXPRESSION})
+    ENDIF()
+    VERIFY_EMPTY(KOKKOS_ADD_TEST ${TEST_UNPARSED_ARGUMENTS})
+  endif()
+ENDFUNCTION()
+
+FUNCTION(KOKKOS_ADD_ADVANCED_TEST)
+  if (KOKKOS_HAS_TRILINOS)
+    TRIBITS_ADD_ADVANCED_TEST(${ARGN})
+  else()
+    # TODO Write this
+  endif()
+ENDFUNCTION()
+
+MACRO(KOKKOS_CREATE_IMPORTED_TPL_LIBRARY TPL_NAME)
+  ADD_INTERFACE_LIBRARY(TPL_LIB_${TPL_NAME})
+  TARGET_LINK_LIBRARIES(TPL_LIB_${TPL_NAME} LINK_PUBLIC ${TPL_${TPL_NAME}_LIBRARIES})
+  TARGET_INCLUDE_DIRECTORIES(TPL_LIB_${TPL_NAME} INTERFACE ${TPL_${TPL_NAME}_INCLUDE_DIRS})
+ENDMACRO()
+
+FUNCTION(KOKKOS_TPL_FIND_INCLUDE_DIRS_AND_LIBRARIES TPL_NAME)
+  if (KOKKOS_HAS_TRILINOS)
+    TRIBITS_TPL_FIND_INCLUDE_DIRS_AND_LIBRARIES(${TPL_NAME} ${ARGN})
+  else()
+    CMAKE_PARSE_ARGUMENTS(PARSE 
+      ""
+      ""
+      "REQUIRED_HEADERS;REQUIRED_LIBS_NAMES"
+      ${ARGN})
+
+    SET(_${TPL_NAME}_ENABLE_SUCCESS TRUE)
+    IF (PARSE_REQUIRED_LIBS_NAMES)
+      FIND_LIBRARY(TPL_${TPL_NAME}_LIBRARIES NAMES ${PARSE_REQUIRED_LIBS_NAMES})
+      IF(NOT TPL_${TPL_NAME}_LIBRARIES)
+        SET(_${TPL_NAME}_ENABLE_SUCCESS FALSE)
+      ENDIF()
+    ENDIF()
+    IF (PARSE_REQUIRED_HEADERS)
+      FIND_PATH(TPL_${TPL_NAME}_INCLUDE_DIRS NAMES ${PARSE_REQUIRED_HEADERS})
+      IF(NOT TPL_${TPL_NAME}_INCLUDE_DIRS)
+        SET(_${TPL_NAME}_ENABLE_SUCCESS FALSE)
+      ENDIF()
+    ENDIF()
+    IF (_${TPL_NAME}_ENABLE_SUCCESS)
+      KOKKOS_CREATE_IMPORTED_TPL_LIBRARY(${TPL_NAME})
+    ENDIF()
+    VERIFY_EMPTY(KOKKOS_CREATE_IMPORTED_TPL_LIBRARY ${PARSE_UNPARSED_ARGUMENTS})
+  endif()
+ENDFUNCTION()
+
+MACRO(KOKKOS_TARGET_COMPILE_OPTIONS TARGET)
+if(KOKKOS_HAS_TRILINOS)
+  TARGET_COMPILE_OPTIONS(${TARGET} ${ARGN})
+else()
+  TARGET_COMPILE_OPTIONS(${TARGET} ${ARGN})
+endif()
```

Comment:
Uhhh?

Reply:
These used to be different. And then I changed tribits. And now they are same. Totally logical!